suse_migration_services/prechecks/scc.py

```diff
@@ -0,0 +1,168 @@
+    and report any error this request might come back with
+    """
+    if migration_system:
+        # This check must not be called inside of the
+        # migration system live iso or container image.
+        # It will only be useful on the system to migrate
+        # at install time of the Migration package or on
+        # manual user invocation of suse-migration-pre-checks
+        # prior the actual migration.
+        return
+
+    installed_products = get_installed_products()
+    installed_products['target_base_product'] = get_migration_target()
+
+    if installed_products.get('installed_products') and \
+       installed_products.get('target_base_product'):
+        registration_server = get_scc_url()
+        registration_credentials = get_scc_credentials()
+        if registration_server and registration_credentials:
+            data = {
+                'installed_products':
+                    installed_products['installed_products'],
+                'target_base_product':
+                    installed_products['target_base_product']
+            }
+            headers = {
+                'accept': 'application/json',
+                'Content-Type': 'application/json'
+            }
+            response = None
+            try:
+                uri = '{}/connect/systems/products/offline_migrations'.format(
+                    registration_server
+                )
+                user = registration_credentials['username']
+                secret = registration_credentials['password']
+                response = requests.post(
+                    uri, json=data, auth=(user, secret), headers=headers
+                )
+                offline_migrations = response.json()
+                result_error = offline_migrations.get('error')
+                if result_error:
+                    log.error(result_error)
+            except Exception as issue:
+                log.error(
+                    'Post request to {} failed with {}'.format(
+                        registration_server,
+                        response.text if response else issue
+                    )
+                )
+            return
+    log.info('System not registered, skipped')
+
+
+def get_scc_credentials():
+    scc_credentials = '/etc/zypp/credentials.d/SCCcredentials'
+    result = {}
+    if os.path.isfile(scc_credentials):
+        try:
+            with open(scc_credentials, 'r') as config:
+                for line in config.read().splitlines():
+                    if line.startswith('username'):
+                        result['username'] = line.split('=')[1]
+                    if line.startswith('password'):
+                        result['password'] = line.split('=')[1]
+        except Exception as issue:
+            message = 'Loading {0} failed: {1}: {2}'.format(
+                scc_credentials, type(issue).__name__, issue
+            )
+            log.error(message)
+    return result
+
+
+def get_scc_url():
```

Comment:
It is not necessarily scc, I suggest `get_registration_server_url` as name for the function.

Reply:
yes that's better, thanks and renamed